### ISSUE TITLE
sftp : support the -s option for specifying the remote subsystem (for use with sudo)

### ISSUE THREAD

**Issue by sheepdestroyer** [rclone/rclone#1801]
I connect to my servers with a user which is not root (ssh for root user is disabled) but this user has sudoers rights. 

To be able to use sftp on system files, I normally use sftp with the following option :  -s "sudo /usr/libexec/openssh/sftp-server" ; and that gives me system level access to the server.

It would be nice to be able to do the same with rclone.

**Comment by ncw** [comment]
I'd probably make this a configuration parameter, say `subsystem` rather than a flag - what do you think?

Looks like we'd have to re-implement the ssh.NewClient fuction, but pass in your sudo command rather than "sftp" - that will probably work.

```go
// NewClient creates a new SFTP client on conn, using zero or more option
// functions.
func NewClient(conn *ssh.Client, opts ...ClientOption) (*Client, error) {
	s, err := conn.NewSession()
	if err != nil {
		return nil, err
	}
	if err := s.RequestSubsystem("sftp"); err != nil {
		return nil, err
	}
	pw, err := s.StdinPipe()
	if err != nil {
		return nil, err
	}
	pr, err := s.StdoutPipe()
	if err != nil {
		return nil, err
	}

	return NewClientPipe(pr, pw, opts...)
}
```

**Comment by sheepdestroyer** [comment]
Yes, it would be great as a conf parameter!

**Comment by smac89** [comment]
@sheepdestroyer  may I ask what command you use to initialize the `sftp`? I tried `sftp -s 'sudo -S /usr/lib/ssh/sftp-server' remote:folder`, but this asks for the password in clear text and then nothing happens even after I enter the password.

Without `-S` to sudo the connection fails with:

> sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper
sudo: a password is required

I tried with `sftp -o RequestTTY=force -s 'sudo -S /usr/lib/ssh/sftp-server' remote:folder`, and sftp errors out with:

> Received message too long 1534293348
Ensure the remote shell produces no output for non-interactive sessions.
